src/ln/msgs.rs

```diff
@@ -681,10 +681,13 @@ impl MsgDecodable for OpenChannel {
 			}
 			shutdown_scriptpubkey = Some(Script::from(v[321..321+len].to_vec()));
 		}
-
+		let mut temp_channel_id = [0; 32];
+		temp_channel_id[..].copy_from_slice(&v[32..64]);
 		Ok(OpenChannel {
 			chain_hash: deserialize(&v[0..32]).unwrap(),
-			temporary_channel_id: deserialize(&v[32..64]).unwrap(),
+			temporary_channel_id: temp_channel_id,
+			chain_hash: deserialize(&v[0..32])?,
```

Comment:
Oops, this line was carried over from the previous change. And the next one.

Reply:
haha oops, happens when you edit in the browser, while sitting in a bus.